### PR TITLE
fix(resumable-media): fix failing unit tests by dropping Python 3.4, add Python 3.8

### LINKED ISSUE #117: Unit test suite under Python 3.4 fails if run after other suites
Running the default unit tests session:
```
$ nox -f noxfile.py -s unit_tests
```
results in an error when computing coverage:
<details>
<summary>traceback</summary>

```
Traceback (most recent call last):
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/coverage/data.py", line 293, in read_file
    self.read_fileobj(f)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/coverage/data.py", line 271, in read_fileobj
    data = self._read_raw_data(file_obj)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/coverage/data.py", line 311, in _read_raw_data
    go_away = file_obj.read(len(cls._GO_AWAY))
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/codecs.py", line 313, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x99 in position 99: invalid start byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/bin/py.test", line 10, in <module>
    sys.exit(main())
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/_pytest/config/__init__.py", line 63, in main
    config = _prepareconfig(args, plugins)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/_pytest/config/__init__.py", line 207, in _prepareconfig
    pluginmanager=pluginmanager, args=args
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/_pytest/helpconfig.py", line 94, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/_pytest/config/__init__.py", line 739, in pytest_cmdline_parse
    self.parse(args)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/_pytest/config/__init__.py", line 947, in parse
    self._preparse(args, addopts=addopts)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/_pytest/config/__init__.py", line 903, in _preparse
    early_config=self, args=args, parser=self._parser
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pytest_cov/plugin.py", line 118, in pytest_load_initial_conftests
    plugin = CovPlugin(early_config.known_args_namespace, early_config.pluginmanager)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pytest_cov/plugin.py", line 164, in __init__
    self.start(engine.Central)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pytest_cov/plugin.py", line 186, in start
    self.cov_controller.start()
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/pytest_cov/engine.py", line 187, in start
    self.cov.load()
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/coverage/control.py", line 677, in load
    self.data_files.read(self.data)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/coverage/data.py", line 653, in read
    data.read_file(self.filename)
  File "/home/peter/workspace/google-resumable-media-python/.nox/unit_tests-3-4/lib/python3.4/site-packages/coverage/data.py", line 297, in read_file
    filename, exc.__class__.__name__, exc,
coverage.misc.CoverageException: Couldn't read data from '/home/peter/workspace/google-resumable-media-python/.coverage': UnicodeDecodeError: 'utf-8' codec can't decode byte 0x99 in position 99: invalid start byte
```
</details>

The error does **not** occur if the `.coverage` file is removed and Python 3.4 tests are run in isolation:
```
$ nox -f noxfile.py -s unit_tests-3.4
```

It might very well be related to the fact that officially [pytest](https://pypi.org/project/pytest/) does not support Python 3.4.

A quick workaround would be to drop Python 3.4 support in this library, as it is already the case with many (all?) `google-cloud-python` libraries, unless of course there are good reasons to keep supporting it.

**Update:** As discussed offline, Python 3.4 support is not needed and should be dropped.

### PR DESCRIPTION
**Pull request by plamut**
Fixes #117.

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 GOOGLE_AUTH = 'google-auth >= 0.10.0'
 
 
-@nox.session(python=['2.7', '3.4', '3.5', '3.6', '3.7'])
+@nox.session(python=['3.5', '3.6', '3.7'])
```
Comment (busunkim96) [inline noxfile.py]: Would you mind dropping 3.8 in here as well?

**Comment by crwilcox** [comment]
Before removing python 2.7 from resumable media we would need to remove it from `google-cloud-storag` and `google-cloud-bigquery` as it is one of our helper packages.

**Comment by plamut** [comment]
I can revert dropping Python 2.7, that commit that introduced a byte string template is optional.

Yes, I can add Python 3.8, too.

File: noxfile.py (hunk 1 above)
Comment (plamut) [reply noxfile.py]: Done.

I also bumped the Python versions in other sessions to `3.8`, as I believe this a standard practice (running misc. checks with the latest supported version).

**Pull request merged**